server/models.py

```diff
@@ -1643,3 +1644,93 @@ class CanvasAssignment(Model):
     @property
     def url(self):
         return '{}/assignments/{}'.format(self.canvas_course.url, self.external_id)
+
+#########
+# Files #
+#########
+
+class ExternalFile(Model):
+
+    id = db.Column(db.Integer, primary_key=True)
+    # Bucket/Folder that the file is stored in
+    container = db.Column(db.String(255), nullable=False)
+    filename = db.Column(db.String(1024), nullable=False)
+    object_name = db.Column(db.String(1024), nullable=False)
+
+    is_staff = db.Column(db.Boolean, nullable=False)
```

Comment:
I don't really like the name `is_staff`, it kinda makes it sound like the file is a staff member. Actually, is this column needed at all? Students can only see their partner's files, so the only time it would make a difference is if a student is partnered with a staff member.